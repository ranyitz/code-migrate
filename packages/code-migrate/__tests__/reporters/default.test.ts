import path from 'path';
import {
  resolveFixture,
  runMigrationAndGetOutput,
  sanitizeStacktrace,
} from '../utils';

const sanitizeDynamicContent = (output: string) => {
  output = output.split('\n').slice(2).join('\n');
  return sanitizeStacktrace(output);
};

describe('default reporter', () => {
  it('passing', async () => {
    const fixtures = resolveFixture('transform');
    const migrationFile = path.join(fixtures, 'migration.ts');

    let output = await runMigrationAndGetOutput({
      fixtures,
      migrationFile,
    });

    expect(sanitizeDynamicContent(output)).toMatchSnapshot();
  });

  it('error', async () => {
    const fixtures = resolveFixture('error');
    const migrationFile = path.join(fixtures, 'migration.ts');

    const output = await runMigrationAndGetOutput({
      fixtures,
      migrationFile,
    });

    expect(sanitizeDynamicContent(output)).toMatchSnapshot();
  });
});
