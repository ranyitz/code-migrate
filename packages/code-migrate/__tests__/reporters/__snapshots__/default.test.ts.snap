// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default reporter error 1`] = `
"
‚ö†Ô∏è  [31mThe following migration tasks were failed, but you can still migrate the rest ‚ö†Ô∏è[39m

[4m[1mtransform bar to baz in json contents[22m[24m
[0m[7m[1m[31m ERROR [39m[22m[27m[0m baz.json

SyntaxError: Unexpected token } in JSON at position 18
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)
    at function (/path/to/file)

[34mü§∑‚Äç No changes have been made[39m"
`;

exports[`default reporter passing 1`] = `
"
[4m[1mtransform bar to baz in json contents[22m[24m
[0m[7m[1m[32m READY [39m[22m[27m[0m baz.json

[32mThe migration has been completed successfully üéâ[39m"
`;
